test/NuGet.Core.FuncTests/Msbuild.Integration.Test/MsbuildRestoreTaskTests.cs

```diff
@@ -807,6 +807,56 @@ public void MsbuildRestore_StaticGraphEvaluation_HandlesInvalidProjectFileExcept
             }
         }
 
+        [PlatformTheory(Platform.Windows)]
+        [InlineData(true)]
+        [InlineData(false)]
+        public void MsbuildRestore_WithMissingProjectReferences_HandlesProjectReferencesToUnsupportedProjects(bool restoreUseStaticGraphEvaluation)
+        {
+            // Arrange
+            using (var pathContext = new SimpleTestPathContext())
+            {
+                // Set up solution, project, and packages
+                var solution = new SimpleTestSolutionContext(pathContext.SolutionRoot);
+
+                NuGetFramework targetFramework = NuGetFramework.Parse("net461");
+
+                var projectA = new SimpleTestProjectContext("ProjectA", ProjectStyle.PackageReference, pathContext.SolutionRoot)
+                {
+                    ToolingVersion15 = true
+                };
+
+                var projectB = new SimpleTestProjectContext("b", ProjectStyle.PackageReference, pathContext.SolutionRoot);
+
+                var projectAFrameworkContext = new SimpleTestProjectFrameworkContext(targetFramework);
+
+                projectAFrameworkContext.ProjectReferences.Add(projectB);
+
+                var packageX = new SimpleTestPackageContext()
+                {
+                    Id = "x",
+                    Version = "1.0.0"
+                };
+
+                projectA.Frameworks.Add(projectAFrameworkContext);
+
+                projectA.AddPackageToAllFrameworks(packageX);
+
+                solution.Projects.Add(projectA);
+                solution.Create(pathContext.SolutionRoot);
+
+                File.WriteAllText(
+                    projectB.ProjectPath,
+                    @"<Project />");
+
+                var result = _msbuildFixture.RunMsBuild(pathContext.WorkingDirectory, $"/t:restore /p:RestoreUseStaticGraphEvaluation={restoreUseStaticGraphEvaluation} {projectA.ProjectPath}", ignoreExitCode: true);
+
+                // Assert
+                result.ExitCode.Should().Be(0, result.AllOutput);
+
+                //result.AllOutput.Should().Contain($"error MSB4025: The project file could not be loaded. Could not find file '{projectB.ProjectPath}'");
```

Comment:
I'm guessing you meant to remove this and the line above?

Reply:
Wow yeah I guess I sent this too soon.  I also figured out why the `.ToArray()` was in `RemoveMissingProjectReferences()`, you can't remove items from a collection that you're iterating over.  So I've updated it to look in reverse order so items can be removed.